src/app/wallet/overview/widgets/coldstake/coldstake.service.ts

```diff
@@ -0,0 +1,108 @@
+import { Injectable, OnDestroy } from '@angular/core';
+import { Observable } from 'rxjs/Observable';
+
+import { Log } from 'ng2-logger';
+import { Amount } from '../../../../core/util/utils';
+
+import { RpcService, RpcStateService } from 'app/core/core.module';
+
+@Injectable()
+export class ColdstakeService implements OnDestroy {
+
+  private destroyed: boolean = false;
+  private log: any = Log.create('coldstake-service');
+
+  public coldstake: any =  {
+    txs: [],
+    amount: 0
+  };
+
+  public hotstake: any = {
+    txs: [],
+    amount: 0
+  };
+
+  coldStakingEnabled: boolean = undefined;
+  walletInitialized: boolean = undefined;
+  public encryptionStatus: string = 'Locked';
+
+  private progress: Amount = new Amount(0, 2);
+  get coldstakeProgress(): number { return this.progress.getAmount() }
+  constructor(
+    private _rpc: RpcService,
+    private _rpcState: RpcStateService
+  ) {
+
+    this._rpcState.observe('getwalletinfo', 'encryptionstatus')
```

Comment:
To update the cold staking widget in a more 'consistent' way, you can also subscribe to 'getwalletinfo'->'txcount', that way on receive of a transaction/zap/etc it will also update the cold staking widget.

Reply:
@kewde but when you do the zap then also it needs to be show the progress to 100% but it doesn't show that after some time like more than 30sec ?